### PR TITLE
docs: document the session name option for multiple sessions

### LINKED ISSUE #969: session utils don't mention anything about the `name` in the config
### Describe the change

I don't mean that it _accepts_ a `name`, but rather that the implementation is built in such a way to support multiple sessions at the same time
```ts
const session = await useSession(event, { name: 'foo', ..})
const session2 = await useSession(event, { name: 'bar, ..})

session.id !== session2
```

This is very useful for holding different references for different things without bloating the very very restrictive space a cookie can hold.
Most commonly, this can be used for flashing data. Something I had to enhance my unstorage driver layers can be done for free. It's a shame really..

Is there a reason for that? If so, is there a reason why it is a map at all?

### URLs

https://h3.unjs.io/examples/handle-session

https://h3.unjs.io/utils/advanced#session-utils

### Additional information

- [ ] Would you be willing to help?

### PR DESCRIPTION
**Pull request by harshagarwalnyu**
Closes #969.

The session example already uses the `name` option (`name: "my-session"`) but never explains what it does, and the ability to run multiple independent sessions isn't documented anywhere. #969 asks for exactly this.

This adds two things to `docs/4.examples/handle-session.md`:

- A short explanation after the existing Options example: `name` sets the cookie (and the `x-{name}-session` header) and defaults to `h3`, which is why the earlier examples produce a cookie named `h3`.
- A "Use Multiple Sessions" section showing that distinct names give you independent sessions stored in separate cookies, with a note that reusing a name reuses the same cookie.

Verified against the current `SessionConfig` on `main` (`name?: string`, default `h3`, also used to derive the `x-{name}-session` header). Docs only, no code changes.

## Summary by CodeRabbit

* **Documentation**
  * Stronger warning: `password` is required for session encryption; avoid hardcoding and load a strong (32+ character) secret from an environment variable.
  * Clarified session options: all are optional except `password`; `name` controls the session cookie and the derived request header (lowercased).
  * Added “Use Multiple Sessions” guidance: sessions with different `name` values can coexist without interference; reusing a `name` shares the cookie (last-write-wins).

### CODE REVIEW

**Comment by harshagarwalnyu** [comment]
Pushed a small fix for the header placeholder point: it now reads `x-<name>-session` and ties it to the concrete `x-h3-session` header already shown earlier in the same page, so it's clear it's a pattern rather than a literal header name.

On the password comments: I kept the literal password value on purpose to stay consistent with the rest of this page. Every existing example here (Initialize, Get Data, Add Data, Clear, Options) uses the same `80d42cfb-...` literal, so swapping only the two new examples to `process.env.SESSION_PASSWORD` would make the file inconsistent, and converting all of them is out of scope for a docs PR that's just about the `name` option. Happy to do a follow-up that moves the whole page to an env-var placeholder + a "use a strong secret, don't commit it" note if you'd prefer that as a separate change.

File: docs/4.examples/handle-session.md [hunk 1]
```diff
@@ -133,3 +133,33 @@ app.use(async (event) => {
   return session.data;
 });
 ```
+
+Every option is optional except `password`. The `name` option is worth calling out: it sets the cookie (and the `x-{name}-session` header) used to store the session, and defaults to `h3`. That default is why the earlier examples set a cookie named `h3`.
```
[the remark this answers is not in the document]
Comment (harshagarwalnyu) [reply docs/4.examples/handle-session.md]: Fixed in 1924aa6. That line now reads `x-<name>-session` and ties it to the concrete `x-h3-session` header the earlier examples produce, so it can no longer be misread as a literal header name.

File: docs/4.examples/handle-session.md [hunk 2]
```diff
@@ -133,3 +133,33 @@ app.use(async (event) => {
...
+
+Because each session is stored under its own `name`, you can run several independent sessions on the same request. They live in separate cookies and never overwrite each other, which is useful for keeping unrelated concerns apart, such as a long-lived auth session and a short-lived flash message:
+
+```js
+import { useSession } from "h3";
+
+app.use(async (event) => {
+  const auth = await useSession(event, {
+    name: "auth",
+    password: "80d42cfb-1cd2-462c-8f17-e3237d9027e9",
```
[the remark this answers is not in the document]
Comment (harshagarwalnyu) [reply docs/4.examples/handle-session.md:152]: Addressed in 7567774. Every example on this page deliberately shares one literal password so the snippets stay copy-paste consistent, so instead of swapping just these two lines (which would leave them out of step with the five identical literals above), I expanded the password `> [!WARNING]` at the top of the page. It now says the hardcoded value is only for readability and that a real app should load it from `process.env.SESSION_PASSWORD`, keep it a strong secret of at least 32 characters, and never commit it. That covers the copy-paste-to-production risk for every example at once.

File: docs/4.examples/handle-session.md [hunk 3]
```diff
@@ -133,3 +133,33 @@ app.use(async (event) => {
...
+  });
+
+  const flash = await useSession(event, {
+    name: "flash",
+    password: "80d42cfb-1cd2-462c-8f17-e3237d9027e9",
```
[the remark this answers is not in the document]
Comment (harshagarwalnyu) [reply docs/4.examples/handle-session.md:157]: Addressed in 7567774. Every example on this page deliberately shares one literal password so the snippets stay copy-paste consistent, so instead of swapping just these two lines (which would leave them out of step with the five identical literals above), I expanded the password `> [!WARNING]` at the top of the page. It now says the hardcoded value is only for readability and that a real app should load it from `process.env.SESSION_PASSWORD`, keep it a strong secret of at least 32 characters, and never commit it. That covers the copy-paste-to-production risk for every example at once.

**Review by pi0x — APPROVED** [review]
**Note: this review was performed by AI agents (Claude Code) on behalf of the maintainer, in two independent passes.**

Pass 1 reviewed the diff for correctness, edge cases, and test coverage. Pass 2 (a fresh agent, adversarial by design) independently re-read the diff, checked for merge conflicts/CI status, and tried to find a reason to block approval before this review was posted. Both passes agree this change is correct, adequately tested, and safe to merge.

Caveat: this repo's real CI workflow (lint + build + vitest) requires maintainer approval to run on fork PRs and has not executed on this PR yet — only automated third-party checks (CodeRabbit, Socket Security) have run. Please trigger/approve the CI workflow before merging.

**Review by pi0 — APPROVED** [review]
Thanks
